sdk/storage_blobs/src/blob/operations/get_blob.rs

```diff
@@ -1,125 +1,129 @@
 use crate::{blob::Blob, prelude::*};
 use azure_core::{
-    error::{ErrorKind, ResultExt},
-    headers::{self, *},
+    collect_pinned_stream,
+    error::{Error, ErrorKind, ResultExt},
+    headers::*,
     prelude::*,
-    CollectedResponse, RequestId,
+    Context, Pageable, RequestId, Response as AzureResponse,
 };
 use bytes::Bytes;
 use chrono::{DateTime, Utc};
-use futures::stream::Stream;
-use std::{
-    convert::{TryFrom, TryInto},
-    str::FromStr,
-};
+use std::str::FromStr;
+
+const DEFAULT_CHUNK_SIZE: u64 = 0x1000 * 0x1000;
 
-#[derive(Debug, Clone, Copy)]
-pub struct GetBlobBuilder<'a> {
-    blob_client: &'a BlobClient,
+#[derive(Clone)]
+pub struct GetBlobBuilder {
+    blob_client: BlobClient,
     range: Option<Range>,
-    blob_versioning: Option<&'a BlobVersioning>,
-    client_request_id: Option<&'a ClientRequestId>,
+    blob_versioning: Option<BlobVersioning>,
     timeout: Option<Timeout>,
     lease_id: Option<LeaseId>,
+    chunk_size: u64,
+    context: Context,
+}
+
```

Comment:
Can we move these helper functions towards the end? It's nice to not have to scroll to see the `impl` block